modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/other/APIM720GetAllEndPointsTestCase.java

```diff
@@ -88,80 +88,62 @@ public APIM720GetAllEndPointsTestCase(TestUserMode userMode) {
+        for (int i = 0; i < 3; i++) {
+            prodEndpointList.add(endpointProduction + "_" + i);
+            sandboxEndpointList.add(endpointSandbox + "_" + i);
+        }
 
-        HttpResponse apiCreateResponse = apiPublisher.addAPI(apiCreationRequestBean);
-        assertEquals(apiCreateResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                     "Invalid Response Code");
+        APIRequest apiRequest = new APIRequest(apiName, apiContext, apiVersion, prodEndpointList, sandboxEndpointList);
+        apiRequest.setEndpointType(endPointType);
+        apiRequest.setTags(tags);
+        apiRequest.setTier(tier);
+        apiRequest.setDescription(description);
+        apiRequest.setVisibility(APIDTO.VisibilityEnum.PUBLIC.getValue());
+        apiIdentifier = new APIIdentifier(apiProvider, apiName, apiVersion);
 
-        //assert JSON object
-        JSONObject createApiJsonObject = new JSONObject(apiCreateResponse.getData());
-        assertEquals(createApiJsonObject.getBoolean("error"), false, "Error in API Creation");
+        HttpResponse apiCreateResponse = apiPublisher.addAPI(apiRequest);
+        assertEquals(apiCreateResponse.getResponseCode(), Response.Status.CREATED.getStatusCode(), "Error when creating API: "
+                + apiName);
+        apiID = apiCreateResponse.getData();
 
-        HttpResponse verifyApiResponse = apiPublisher.getApi(apiName, apiProvider, apiVersion);
-        JSONObject verifyApiJsonObject = new JSONObject(verifyApiResponse.getData());
-        assertFalse(verifyApiJsonObject.getBoolean("error"), "Error in Verify API Response");
+        HttpResponse getAPIResponse = apiPublisher.getAPI(apiID);
+        assertEquals(getAPIResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
+                "Error when retrieving API: " + apiName);
 
         //publish API
-        APILifeCycleStateRequest updateRequest = new APILifeCycleStateRequest(apiName, apiProvider,
-                                                                              APILifeCycleState.PUBLISHED);
+        HttpResponse changeLCStatusResponse = apiPublisher.changeAPILifeCycleStatus(apiID,
+                APILifeCycleAction.PUBLISH.getAction(), null);
 
+        assertEquals(changeLCStatusResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
+                     "Error when publishing the API: " + apiName);
 
-        HttpResponse statusUpdateResponse = apiPublisher.changeAPILifeCycleStatus(updateRequest);
-        assertEquals(statusUpdateResponse.getResponseCode(), Response.Status.OK.getStatusCode(),
-                     "Response Code is Mismatched");
-
-        waitForAPIDeploymentSync(apiProvider, apiName, apiVersion, APIMIntegrationConstants.IS_API_EXISTS);
-
-        JSONObject statusUpdateJsonObject = new JSONObject(statusUpdateResponse.getData());
-        assertFalse(statusUpdateJsonObject.getBoolean("error"), "API is not published");
-
+        Thread.sleep(15000);
```

Comment:
Why have we added this sleep? This is not the best practice.

Reply:
Fixed